src/platform/api/types.ts

```diff
@@ -152,3 +153,8 @@ export const ILanguageServerProvider = Symbol('ILanguageServerProvider');
 export interface ILanguageServerProvider {
     getLanguageServer(resource?: InterpreterUri): Promise<ILanguageServer | undefined>;
 }
+
+export const IExportedKernelServiceFactory = Symbol('IExportedKernelServiceFactory');
+export interface IExportedKernelServiceFactory {
+    getService(): Promise<IExportedKernelService | undefined>;
```

Comment:
To make this work for web, I split this out into a service factory. This way we can have a web and a node version of the JupyterKernelServiceFactory.